code/formatters/spdx.py

```diff
@@ -89,19 +96,29 @@ def format_report(cls, report: DocumentReport):
             finding_annotation.annotation_type = "OTHER"
             finding_annotation.comment = finding.json()
             SPDX_document.add_annotation(finding_annotation)
+
         dependent_documents = []
         for dependency in report.dependencies:
             dep_document = create_base_spdx_document(dependency.name)
-            dep_document.package = Package(name=dependency.name, spdx_id=dep_document.spdx_id, download_location=dependency.download_location, version=dependency.version)
+            dep_document.package = Package(name=dependency.name, spdx_id=dep_document.spdx_id,
+                                            download_location=dependency.download_location, version=dependency.version)
+            checksums = create_spdx_namespace(dependency.name)
+            # set SHA1 checksum
+            dep_document_checksum = Algorithm(identifier="SHA1", value=checksums['sha1'])
+            Algorithm(identifier="SHA-512", value=hashlib.sha512(cls.__TO_METHOD__(dep_document).encode()).hexdigest())
+            # set SHA512 checksum
+            dep_document_checksum.add_algorithm(Algorithm(identifier="SHA512", value=checksums['sha512']))
```

Comment:
I do not believe the `spdx.checksum.Algorithm` class has an "add_algorithm" model. Please confirm that this code is correct.